specification/billing/resource-manager/Microsoft.Billing/stable/2023-04-01/billingProperty.json

```diff
@@ -65,182 +95,553 @@
+      "description": "A billing property.",
+      "type": "object",
       "allOf": [
         {
           "$ref": "./types.json#/definitions/ArmResource"
         }
       ],
-      "description": "A billing property.",
       "properties": {
         "properties": {
-          "$ref": "#/definitions/BillingPropertyProperties",
-          "description": "A billing property.",
+          "description": "The properties of a(n) BillingProperty",
+          "allOf": [
+            {
+              "$ref": "#/definitions/BillingPropertyProperties"
+            }
+          ],
           "x-ms-client-flatten": true
         }
       },
-      "type": "object"
+      "x-ms-azure-resource": true
     },
     "BillingPropertyProperties": {
-      "description": "The billing property.",
+      "description": "A billing property.",
+      "type": "object",
       "properties": {
-        "accountAdminNotificationEmailAddress": {
-          "description": "The email address on which the account admin gets all Azure notifications.",
+        "billingAccountAgreementType": {
+          "description": "The type of agreement.",
+          "enum": [
+            "Unknown",
+            "MicrosoftCustomerAgreement",
+            "EnterpriseAgreement",
+            "MicrosoftOnlineServicesProgram",
+            "MicrosoftPartnerAgreement"
+          ],
+          "type": "string",
           "readOnly": true,
-          "type": "string"
+          "x-ms-enum": {
+            "name": "AgreementType",
+            "modelAsString": true
+          }
         },
         "billingAccountDisplayName": {
-          "description": "The name of the billing account to which the subscription is billed.",
-          "readOnly": true,
-          "type": "string"
+          "description": "The name of the billing account.",
+          "type": "string",
+          "readOnly": true
         },
         "billingAccountId": {
-          "description": "The ID of the billing account to which the subscription is billed.",
+          "description": "The fully qualified ID that uniquely identifies a billing account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountNotificationEmailAddress": {
+          "description": "Notification email address for legacy account. Available for agreement type Microsoft Online Services Program.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountSoldToCountry": {
+          "description": "The address of the individual or organization that is responsible for the billing account.",
+          "type": "string",
+          "readOnly": true
+        },
+        "billingAccountStatus": {
+          "description": "The current status of the billing account.",
+          "enum": [
+            "Unknown",
+            "Active",
+            "UnderReview",
+            "Disabled",
+            "Deleted",
+            "Extended",
+            "Pending",
+            "New"
```

Comment:
At what point do we move from New to Active?

Reply:
Should we even have New?

Reply:
@ramkd12 @amberbhargava to comment here